### PR TITLE
Working branch5

### PR DESCRIPTION
**Pull request by vflorin1995**
### Changes

- Extract some pieces of logic into separate classes
- Main.rb is creating the entry point only

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  def initialize(date, book, person)
+    @date = date
+
+    @book = book
+    book.rental << self
+
+    @person = person
+    person.rental << self
+  end
+end
```
Comment (Munsa1) [inline rental.rb:13]: - [ ] Currently the user is unable to list rentals using `id`, kindly fix this as it is a requirement
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @vflorin1995  @Joel-100 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 1 above)
Comment (vflorin1995) [reply rental.rb:13]: @Munsa1 The person ID is not 1. You can see the person ID if you display all persons.

**Review by medaminedev66 — COMMENTED** [review]
Hello @vflorin1995 👋,

_This is an additional review as a result of the second opinion request._

I tested your app, and the user can list rentals using the person's id. Great work @vflorin1995 👏 .

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approve right away :)

Keep rocking 🚀!

**Review by t-yanick — APPROVED** [review]
Hi @vflorin1995 and @Joel-100 , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**